src/apps/replication/lib/mutation.h

```diff
@@ -133,13 +133,17 @@ class mutation_queue
     ~mutation_queue()
     {
         clear();
+        // TODO(qinzuoyan): if primary writing is failed (prepare failure or writing log failure),
```

Comment:
right, so condition is too strong. it was deleted in other repos and will be pushed later.